runtime/parser/declaration.go

```diff
@@ -957,6 +1025,174 @@ func parseFieldWithVariableKind(
 	), nil
 }
 
+// parseEntitlementMapping parses an entitlement mapping
+//
+//	entitlementMapping : nominalType '->' nominalType
+func parseEntitlementMapping(p *parser, docString string) (*ast.EntitlementMapElement, error) {
+	inputType, err := parseType(p, lowestBindingPower)
+	inputNominalType, ok := inputType.(*ast.NominalType)
+	if !ok {
+		p.reportSyntaxError(
+			"expected nominal type, got %s",
+			inputType,
+		)
+	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Shouldn't the error-check be moved to right after parsing the type?

```suggestion
	inputType, err := parseType(p, lowestBindingPower)
	if err != nil {
		return nil, err
	}

	inputNominalType, ok := inputType.(*ast.NominalType)
	if !ok {
		p.reportSyntaxError(
			"expected nominal type, got %s",
			inputType,
		)
	}
```